offer-manager-domain/src/main/groovy/life/qbic/business/customers/update/UpdateCustomer.groovy

```diff
@@ -1,10 +1,54 @@
 package life.qbic.business.customers.update
 
+import life.qbic.datamodel.dtos.business.Customer
+
+import life.qbic.business.exceptions.DatabaseQueryException
+
 /**
- * A project manager wants to modify the customer information
+ * This use case updates an existing customer in the system. New Affiliations of the customer are added to the respective table.
+ * If other changes are made to the customer, a new customer is created in the system and the old customer is set to inactive.
  *
  * @since: 1.0.0
- * @author: Tobias Koch
+ * @author: Andreas Friedrich, Tobias Koch
  */
-class UpdateCustomer {
+class UpdateCustomer implements UpdateCustomerInput {
+
+  private UpdateCustomerDataSource dataSource
+  private UpdateCustomerOutput output
+
+  UpdateCustomer(UpdateCustomerOutput output, UpdateCustomerDataSource dataSource){
+    this.output = output
+    this.dataSource = dataSource
+  }
+
+  @Override
+  void updateCustomer(String customerId, Customer customer) {
+    Customer existingCustomer = dataSource.getCustomer(Integer.parseInt(customerId))
+
+    boolean noFundamentalChange = existingCustomer.firstName.equals(customer.firstName)
+    && existingCustomer.lastName.equals(customer.lastName)
+    && existingCustomer.emailAddress.equals(customer.emailAddress)
+    && existingCustomer.title.equals(customer.title)
+
+    try {
+      if(noFundamentalChange) {
+        dataSource.updateCustomerAffiliations(customerId, customer.affiliations)
```

Comment:
Why would you provide a list of affiliations instead of a single affiliation?

Reply:
We discussed this now, but for anyone following at home: the customer object contains a list of affiliations of which one may be new. It is easier to handle the use case this way and re-use functionality in the database connector.